private/pkg/protosource/file.go

```diff
@@ -244,16 +244,18 @@ func newFile(inputFile InputFile) (*file, error) {
 	f.descriptor = descriptor
 
 	if inputFile.IsSyntaxUnspecified() {
-		f.syntax = SyntaxUnspecified
-	} else {
-		switch syntaxString := f.fileDescriptor.GetSyntax(); syntaxString {
 		// if the syntax is "proto2", protoc and buf will not set the syntax
 		// field even if it was explicitly set, this is why we have
 		// IsSyntaxUnspecified
+		f.syntax = SyntaxUnspecified
+	} else {
+		switch syntaxString := f.fileDescriptor.GetSyntax(); syntaxString {
 		case "", "proto2":
 			f.syntax = SyntaxProto2
 		case "proto3":
 			f.syntax = SyntaxProto3
+		case "editions":
+			f.syntax = SyntaxEditions
```

Comment:
We should not allow this. This case is only valid when the  `edition` field is set to a valid edition, and there are no known/valid editions yet.

Reply:
`protosource` is not a compiler, it assumes it is being given a valid `FileDescriptorSet`. Given that assumption, and that a value of syntax can be `editions`, we should handle it.